### ISSUE TITLE
$ref in external file in parameter with resolve option true

### ISSUE THREAD

**Issue by jmini** [swagger-api/swagger-parser#749]
When `$ref` references an external yaml file, Swagger-Parser can merge everything and produce an instance of OpenAPI where the references are local.
The key point is to use `setResolve(true)`.
See more: https://j2r2b.github.io/2018/05/26/external-ref-openapi.html

It was reported that this does not work with `$ref` in parameter. It works well in request body
and response.

---

Test case, in the same folder, put two files:

Main OAS3 file: `main.yaml`

```yaml
openapi: 3.0.1
info:
  title: ping test
  version: '1.0'
servers:
  - url: 'http://localhost:8000/'
paths:
  /some/ping:
    get:
      operationId: pingGet
      parameters:
        - name: i
          in: query
          description: Test
          required: true
          schema:
            $ref: './external.yaml#/SomeId'
      responses:
        '201':
          description: OK
components:
  schemas: {}
```

External file: `other.yaml`

```yaml
SomeId:
  type: integer
  format: int32
  description: My value 123
  example: 123
```

Run this code:

```
OpenAPIParser openApiParser = new OpenAPIParser();
ParseOptions options = new ParseOptions();
options.setResolve(true);

OpenAPI openAPI = openApiParser.readLocation(inputSpec, null, options).getOpenAPI();
String string = Yaml.mapper().writerWithDefaultPrettyPrinter().writeValueAsString(openAPI);
```

Current value:
```yaml
openapi: 3.0.1
info:
  title: ping test
  version: "1.0"
servers:
- url: http://localhost:8000/
paths:
  /some/ping:
    get:
      operationId: pingGet
      parameters:
      - name: i
        in: query
        description: Test
        required: true
        style: form
        explode: true
        schema:
          $ref: ./external.yaml#/SomeId
      responses:
        201:
          description: OK
components:
  schemas: {}
```

Expected value:

```yaml
openapi: 3.0.1
info:
  title: ping test
  version: "1.0"
servers:
- url: http://localhost:8000/
paths:
  /some/ping:
    get:
      operationId: pingGet
      parameters:
      - name: i
        in: query
        description: Test
        required: true
        style: form
        explode: true
        schema:
          $ref:  '#/components/schemas/SomeId'
      responses:
        201:
          description: OK
components:
  schemas:
    SomeId:
      type: integer
      description: My value 123
      format: int32
      example: 123
```

Tested with version `2.0.1`

**Comment by ymohdriz** [comment]
Hi team,

I have raised a PR (https://github.com/swagger-api/swagger-parser/pull/752) to fix this issue.

Thanks,
Mohammed

**Comment by gracekarina** [comment]
Fixed by #752
